nodeup/pkg/model/docker.go

```diff
@@ -435,76 +114,6 @@ func (b *DockerBuilder) Build(c *fi.ModelBuilderContext) error {
 		return err
 	}
 
-	// Add packages
-	{
-		count := 0
-		for i := range dockerVersions {
-			dv := &dockerVersions[i]
-			if !dv.matches(b.Architecture, dockerVersion, b.Distribution) {
-				continue
-			}
-
-			count++
-
-			var packageTask fi.Task
-			if dv.PlainBinary {
-				packageTask = &nodetasks.Archive{
-					Name:      "docker-ce",
-					Source:    dv.Source,
-					Hash:      dv.Hash,
-					MapFiles:  dv.MapFiles,
-					TargetDir: "/",
-				}
-				c.AddTask(packageTask)
-
-				c.AddTask(b.buildDockerGroup())
-				c.AddTask(b.buildSystemdSocket())
-			} else {
-				var extraPkgs []*nodetasks.Package
-				for name, pkg := range dv.ExtraPackages {
-					dep := &nodetasks.Package{
-						Name:         name,
-						Version:      s(pkg.Version),
-						Source:       s(pkg.Source),
-						Hash:         s(pkg.Hash),
-						PreventStart: fi.Bool(true),
-					}
-					extraPkgs = append(extraPkgs, dep)
-				}
-				packageTask = &nodetasks.Package{
-					Name:    dv.Name,
-					Version: s(dv.Version),
-					Source:  s(dv.Source),
-					Hash:    s(dv.Hash),
-					Deps:    extraPkgs,
-
-					// TODO: PreventStart is now unused?
-					PreventStart: fi.Bool(true),
-				}
-				c.AddTask(packageTask)
-			}
-
-			// As a mitigation for CVE-2019-5736 (possibly a fix, definitely defense-in-depth) we chattr docker-runc to be immutable
-			for _, f := range dv.MarkImmutable {
-				c.AddTask(&nodetasks.Chattr{
-					File: f,
-					Mode: "+i",
-					Deps: []fi.Task{packageTask},
-				})
-			}
```

Comment:
Do we not need this mitigation anymore?

Reply:
I saw this done only for 17.03.x and didn't seem like something that important anymore considering the number of security issues with that version. Anyone interested in security should really use a newer Docker version.

Reply:
It doesn't have value for newer versions, but I don't think we should reintroduce a significant security vulnerability for 17.03.

Perhaps we should work on our deprecation policy for container runtimes.

Reply:
Good call on digging a bit more. Seems that this is wider than initially thought and affects about anything < 18.09.2.
Re-added the mitigation and updated the tests output.

> Perhaps we should work on our deprecation policy for container runtimes.
+1 from me 😄